docs_src/design/adr/0026-Common Configuration.md

```diff
@@ -0,0 +1,393 @@
+
+[Writable]
+  # InsecureSecrets are required for when Redis is used for message bus
+  [Writable.InsecureSecrets]
+    [Writable.InsecureSecrets.MQTT]
+    path = "credentials"
+      [Writable.InsecureSecrets.MQTT.Secrets]
+      username = ""
+      password = ""
+
+[Service]
+Port = 59982
+StartupMsg = "device mqtt started"
+
+[Clients]
+  [Clients.core-data]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 59880
+
+  [Clients.core-metadata]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 59881
+
+[MessageQueue]
+PublishTopicPrefix = "edgex/events/device" # /<device-profile-name>/<device-name>/<source-name> will be added to this Publish Topic prefix
+  [MessageQueue.Optional]
+  # Default MQTT & NATS Specific options that need to be here to enable environment variable overrides of them
+  ClientId = "device-mqtt"
+  [MessageQueue.Topics]
+  CommandRequestTopic = "edgex/device/command/request/device-mqtt/#"   # subscribing for inbound command requests
+  CommandResponseTopicPrefix = "edgex/device/command/response"   # publishing outbound command responses; <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+
+[MQTTBrokerInfo]
+Schema = "tcp"
+Host = "localhost"
+Port = 1883
+Qos = 0
+KeepAlive = 3600
+ClientId = "device-mqtt"
+
+CredentialsRetryTime = 120 # Seconds
+CredentialsRetryWait = 1 # Seconds
+ConnEstablishingRetry = 10
+ConnRetryWaitTime = 5
+
+# AuthMode is the MQTT broker authentication mechanism. Currently, "none" and "usernamepassword" is the only AuthMode supported by this service, and the secret keys are "username" and "password".
+AuthMode = "none"
+CredentialsPath = "credentials"
+
+# Comment out/remove when using multi-level topics
+IncomingTopic = "DataTopic"
+ResponseTopic = "ResponseTopic"
+UseTopicLevels = false
+
+# Uncomment to use multi-level topics
+# IncomingTopic = "incoming/data/#"
+# ResponseTopic = "command/response/#"
+# UseTopicLevels = true
+
+    [MQTTBrokerInfo.Writable]
+    # ResponseFetchInterval specifies the retry interval(milliseconds) to fetch the command response from the MQTT broker
+    ResponseFetchInterval = 500
+```
+
+### Modules and Services Impacted
+
+The following modules and services are impacted:
+
+- go-mod-bootstrap
+    - All service configuration bootstrapping is handled in this module
+- All services which consume go-mod-bootstrap
+    - The configuration TOML files will be updated to remove all common  settings
+    - This leaves only the private settings and overrides of certain common settings (See examples above)
+- New service which provides the common configuration
+    - Contains new configuration TOML which contains all the common settings (See example above) 
+    - No environment overrides will be applied to the common settings
+    - This service is responsible for 
+      - Pushing the common settings into the Configuration Provider during first start-up

```

Comment:
May be obvious, but this service also has the command line "override" feature so that the common settings can be updated/replaced in the CP?

Reply:
Hadn't thought about it, but make sense to include. THX!

Reply:
> Suggest the ADR specifies the new environment variables related to secret store and common config locations

@iain-anderson 

added as discussed in meeting: https://github.com/edgexfoundry/edgex-docs/pull/909/commits/3307591538561cfe40e1f739a2cfefe096121910#diff-bf54438fd0375b5394a29d2beb1dea70d30da1231f568749d7dca48b8f54e4c4R42

Reply:
done: https://github.com/edgexfoundry/edgex-docs/pull/909/commits/3307591538561cfe40e1f739a2cfefe096121910#diff-bf54438fd0375b5394a29d2beb1dea70d30da1231f568749d7dca48b8f54e4c4R377